src/storage/mongodb/SiteAreaStorage.ts

```diff
@@ -253,11 +277,43 @@ export default class SiteAreaStorage {
         asField: 'site', oneToOneCardinality: true
       });
     }
+    // Site Area Parent
+    if (params.withParentSiteArea) {
+      DatabaseUtils.pushSiteAreaLookupInAggregation({
+        tenantID: tenant.id, aggregation, localField: 'parentSiteAreaID', foreignField: '_id',
+        asField: 'parentSiteArea', oneToOneCardinality: true
+      });
+    }
     // Charging Stations
     if (params.withChargingStations || params.withOnlyChargingStations || params.withAvailableChargingStations) {
+      let pipelineMatch = {};
+      const additionalPipeline = [];
+      if (!Utils.isEmptyArray(params.chargingStationConnectorStatuses)) {
```

Comment:
I think that Charging Station's statuses should be done before the sort/skip/limit code like all the filters.